Added demo code for a Classifier and Transformer

vighneshbirodkar opened the pull request:
Addresses #10 
The code for the 1-NN classifier was taken from 
https://github.com/scikit-learn/scikit-learn/issues/6526

/ping @MechCoder (my best friend)
@mblondel

vighneshbirodkar committed:
Added TemplateClassifer

vighneshbirodkar committed:
Added TemplateTransformer

MechCoder commented on skltemplate/template.py:
```diff
@@ -47,4 +51,135 @@ def predict(self, X):
+        # Check that X and y have correct shape
+        X, y = check_X_y(X, y)
+        # Store the classes seen durind fit
+        self.classes_ = unique_labels(y)
```
Using `check_classification_targets` from utils.multiclass should be a better way of testing this, especially since it was made for this specific purpose.

vighneshbirodkar replied:
At least in this case we can't do anything with multiple labels.

MechCoder replied:
I agree that `check_classification_targets` is placed in the wrong module

MechCoder commented on examples/plot_classifier.py:
```diff
@@ -0,0 +1,30 @@
+ax0 = plt.scatter(X_0[:, 0], X_0[:, 1], c='crimson', s=30)
+ax1 = plt.scatter(X_1[:, 0], X_1[:, 1], c='deepskyblue', s=30)
+
+
```
1 extra line

MechCoder commented on examples/plot_classifier.py:
```diff
@@ -0,0 +1,30 @@
+
+ax0 = plt.scatter(X_0[:, 0], X_0[:, 1], c='crimson', s=30)
+ax1 = plt.scatter(X_1[:, 0], X_1[:, 1], c='deepskyblue', s=30)
+
```
Eventhough it is a dummy example, it might be good to just plot the points [0, 0] and [1, 1]

MechCoder commented on examples/plot_transformer.py:
```diff
@@ -0,0 +1,21 @@
+estimator = TemplateTransformer()
+X_transformed = estimator.fit_transform(X)
+
+plt.plot(X.flatten()/X.max(), label='Original Data')
```
space after `/`

MechCoder commented on examples/plot_transformer.py:
```diff
@@ -0,0 +1,21 @@
+
+plt.plot(X.flatten()/X.max(), label='Original Data')
+plt.plot(X_transformed.flatten()/X_transformed.max(), label='Transformed Data')
+plt.title('Scaled plots of original and transformed data')
```
Labels to the axes?

MechCoder commented:
@vighneshbirodkar Thanks! Just some minor cosmetics. Feel free to merge after addressing.

Do you think it would be a good idea to document somewhere what has to be done so that tests pass after inheriting from `ClassifierMixin` and `TransformerMixin`

To me it seems for a classifier, we need to make sure that the target is discrete, fitting on a single label should return the same label during predict. For a transformer, fit_transform should be consistent with fit().transform() etc..

vighneshbirodkar committed:
Made graphs look better

vighneshbirodkar commented:
@MechCoder Can you have one more look ? I have addressed all except one comment. I have added comments in the code as to why each step is necessary. What other documentation you think is necessary ?

MechCoder commented:
Thanks and sorry for the delay
